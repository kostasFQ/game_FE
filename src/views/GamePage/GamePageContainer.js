import { connect } from 'react-redux';
import GamePage from './GamePageComponent';
import { saveCount } from 'store/game/actions';

const mapDispatchToProps = dispatch => ({
  saveCount: count => dispatch(saveCount(count)),
})

const mapStateToProps = state => ({
  game: state.game,
});

export default connect(mapStateToProps, mapDispatchToProps)(GamePage);