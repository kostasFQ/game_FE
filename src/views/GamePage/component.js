import React, { PureComponent } from 'react';
import { withRouter } from 'react-router-dom';
import MainButtom from 'components/Buttons/MainButton/component';
import Counter from 'components/Counter/component';
import Countdown from 'components/Countdown/container';
import FinalCount from 'components/FinalCount/container';
import styles from 'views/assets/GamePage.module.scss';

class GamePage extends PureComponent {
  state = {
    clicks: 0,
  }

  increment = () => {
    this.setState((state) => ({ clicks: state.clicks + 1 }));
  }

  componentDidUpdate(prevProps) {
    const { clicks } = this.state;
    const { saveCount, game: { gameOver } } = this.props;
    const { game: { gameOver: prevGameOver } } = prevProps;

    if (gameOver && prevGameOver !== gameOver) {
      saveCount(clicks);
      this.setState(() => ({ clicks: 0 }));
    }
  }

  render() {
    const { clicks } = this.state;
    const { game: { gameStarted, gameOver } } = this.props;

    if (!gameStarted) { return ( <div className={styles.gamePage__container}> <Countdown /> </div> ) }

    if (gameOver) { return ( <div className={styles.gamePage__container}> <FinalCount /> </div> ) }

    return (
      <div className={styles.gamePage__container}>
        <Counter clicks={clicks} />
        <MainButtom onClick={this.increment} />
      </div>
    );
  }
}

export default withRouter(GamePage);