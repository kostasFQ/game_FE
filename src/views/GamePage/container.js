import { connect } from 'react-redux';
import GamePage from './component';
import { saveCount } from 'store/game/actions';

const mapDispatchToProps = dispatch => ({
  saveCount: count => dispatch(saveCount(count)),
})

const mapStateToProps = state => ({
  game: state.game,
});

export default connect(mapStateToProps, mapDispatchToProps)(GamePage);