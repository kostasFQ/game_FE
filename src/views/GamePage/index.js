import GamePage from './GamePageContainer';

export default GamePage;