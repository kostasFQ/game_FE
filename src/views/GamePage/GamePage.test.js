import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { StaticRouter } from 'react-router-dom';
import _store from '../../../mocks/store';
import GamePage from './container';
const mockStore = configureStore([thunk]);

test('GamePage should be render Countdown', () => {
  const initialState = { game: _store };
  const store = mockStore(initialState);
  const instance = renderer.create(
    <Provider store={store}>
      <StaticRouter location="/game">
        <GamePage />
      </StaticRouter>
    </Provider>,
  );

  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

test('GamePage should be render FinalCount', () => {
  const store2 = {
    ..._store,
    gameStarted: true,
    gameOver: true
  }
  const initialState = { game: store2 };
  const store = mockStore(initialState);
  const instance = renderer.create(
    <Provider store={store}>
      <StaticRouter location="/game">
        <GamePage />
      </StaticRouter>
    </Provider>,
  );

  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

test('GamePage should be render', () => {
  const store3 = {
    ..._store,
    gameStarted: true,
    gameOver: false
  }
  const initialState = { game: store3 };
  const store = mockStore(initialState);
  const instance = renderer.create(
    <Provider store={store}>
      <StaticRouter location="/game">
        <GamePage />
      </StaticRouter>
    </Provider>,
  );

  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

test('GamePage update', () => {
  const initialState = { game: _store };
  const store = mockStore(initialState);
  const instance = renderer.create(
    <Provider store={store}>
      <StaticRouter location="/game">
        <GamePage />
      </StaticRouter>
    </Provider>,
  );

  const _updatedStore = {
    ..._store,
    gameStarted: true,
    gameOver: true
  };
  const _initialState = { game: _updatedStore };
  const updatedStore = mockStore(_initialState);
  instance.update(
    <Provider store={updatedStore}>
      <StaticRouter location="/game">
        <GamePage />
      </StaticRouter>
    </Provider>,
  );

  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})

