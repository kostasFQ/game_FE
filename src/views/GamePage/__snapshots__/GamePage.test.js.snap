// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalCount should be render 1`] = `
<div
  className="gamePage__container"
>
   
  <div>
    start from 
    3
     seconds
  </div>
   
</div>
`;
